Source/SwiftLintFramework/Protocols/Rule.swift

```diff
@@ -32,33 +32,46 @@ public protocol ConfigurableRule: Rule {
     func isEqualTo(rule: ConfigurableRule) -> Bool
 }
 
-public protocol ParameterizedRule: ConfigurableRule {
-    typealias ParameterType: Equatable
-    init(parameters: [RuleParameter<ParameterType>])
-    var parameters: [RuleParameter<ParameterType>] { get }
+public protocol ViolationLevelRule: ConfigurableRule {
+    var warning: RuleParameter<Int> { get set }
```

Comment:
do you need the `set`s here? If you move `self.init()` from the `ViolationLevelRule` extension to happen after setting the `warning` and `error` properties, I don't think you'll need the `set`s here.